src/components/queryBuilder/OrderByEditor.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { SelectableValue } from '@grafana/data';
+import { Button, InlineFormLabel, Select } from '@grafana/ui';
+import {
+  OrderBy,
+  OrderByDirection,
+  QueryBuilderOptions,
+  TableColumn,
+  BuilderMode,
+  AggregateColumn,
+} from 'types/queryBuilder';
+import allLabels from 'labels';
+import { styles } from 'styles';
+
+interface OrderByItemProps {
+  columnOptions: Array<SelectableValue<string>>;
+  index: number,
+  orderByItem: OrderBy;
+  updateOrderByItem: (index: number, orderByItem: OrderBy) => void;
+}
+
+const sortOptions = [
+  { label: 'ASC', value: OrderByDirection.ASC },
+  { label: 'DESC', value: OrderByDirection.DESC },
+];
+
+const OrderByItem = (props: OrderByItemProps) => {
+  const { columnOptions, index, orderByItem, updateOrderByItem } = props;
+
+  return (
+    <>
+      <Select
+        value={orderByItem.name}
+        className={styles.Common.inlineSelect}
+        width={20}
+        options={columnOptions}
+        onChange={e => updateOrderByItem(index, { ...orderByItem, name: e.value! })}
+        allowCustomValue={true}
+        menuPlacement={'bottom'}
+      ></Select>
+      <Select<OrderByDirection>
+        value={orderByItem.dir}
+        className={styles.Common.inlineSelect}
+        width={12}
+        options={sortOptions}
+        onChange={e => updateOrderByItem(index, { ...orderByItem, dir: e.value! })}
+        menuPlacement={'bottom'}
+      />
+    </>
+  );
+};
+
+interface OrderByEditorProps {
+  allColumns: TableColumn[];
+  orderBy: OrderBy[];
+  onOrderByChange: (orderBy: OrderBy[]) => void;
+}
+export const OrderByEditor = (props: OrderByEditorProps) => {
+  const { allColumns, orderBy, onOrderByChange } = props;
+  const columnOptions: Array<SelectableValue<string>> = (allColumns || []).map(c => ({ label: c.name, value: c.name }));
+  const { label, tooltip, addLabel } = allLabels.components.OrderByEditor;
+
+  const addOrderByItem = () => {
```

Comment:
I'd rather extract them in array helpers `(input: ReadonlyArray<T>, ...): ReadonlyArray<T>` it would help us to add tests and reduce the scope of the component.